src/util/projectUtils.js

```diff
@@ -24,3 +28,13 @@ export function saveCurrentProject(state) {
 
   return false;
 }
+
+export function saveProjectWithKey(state, projectKey) {
+  const userId = getCurrentUserId(state);
+  const project = getProject(state, {projectKey});
+  if (userId) {
```

Comment:
Seems like this could also check for pristine-ness? In general this method looks to be 90% the same logic as `saveCurrentProject` so probably best to abstract out the shared logic?

Reply:
Yes indeed!